File: src/controller/controlScript.py
import subprocess
import os
import tkinter
import tkinter.filedialog
import tkinter.messagebox
from pathlib import Path

options = []

class ControScript:
    @staticmethod
    def execScript(scriptPath):
        script_path = Path(scriptPath)

        if not script_path.exists():
            raise FileNotFoundError(f"Arquivo não encontrado: {script_path}")
        
        ext = os.path.splitext(script_path)[1].lower()

        if ext == '.cmd':
            command = f'cmd /c start cmd /k "{script_path}"'
        else:
            command = f'cmd /c start powershell -NoExit -File "{script_path}"'
        
        try:
            result = subprocess.Popen(command, shell=True)
            return result
        except subprocess.CalledProcessError as error:
            tkinter.messagebox.showerror(
                "Erro ao executar o script",
                f"Ocorreu um erro ao executar o script: {error}\n"
                f"Saída do script:\n{error.stdout}"
            )
            return error

    @staticmethod
    def execButton(select):
        try:
            result = ControScript.execScript(select)
            return result
        except KeyError:
            tkinter.messagebox.showinfo("Informação", "Opção inválida selecionada!")
        except FileNotFoundError as error:
            tkinter.messagebox.showerror("Erro", str(error))

    @staticmethod
    def searchButton():
        dir_win = tkinter.filedialog.askopenfilename(title="Selecione um arquivo")

        if dir_win:
            options.append(dir_win)
            ControScript.execButton(dir_win)


        return dir_win

File: src/controller/control_script.rs
use std::process::{Command, Stdio};
use std::path::Path;
use std::fs;
use std::io;
use rfd::FileDialog;
use native_dialog::{MessageDialog, MessageType};
use std::sync::Mutex;

lazy_static::lazy_static! {
    static ref OPTIONS: Mutex<Vec<String>> = Mutex::new(new![]);
}

fn exec_script(script_path: &str) -> Result<(), io::Error>  {
    let path = Path::new(script_path);

    if !path.exists() {
        return Err(io::Error::new(io::ErrorKind::NotFound, "Arquivo não encontrado"));
    }

    let ext = path.extension().and_then(|e| e.to_str()).unwrap_or("").to_lowercase();

    let command = if ext == "cmd" {
        format!("cmd /c start cmd /k \"{}\"", path.display())
    } else {
        format!("cmd /c start powershell -NoExit -File \"{}\"", path.display())
    };

    let result = Command::new("cmd")
        .args(&["/c", &command])
        .stdout(Stdio::inherit())
        .stderr(Stdio::inherit())
        .spawn();

    match result {
        Ok(_) => Ok(()),
        Err(error) => {
            MessageDialog::new()
                .set_type(MessageType::Error)
                .set_title("Erro ao executar script")
                .set_text(&format!("Ocorreu um erro ao executar o script: {}", error))
                .show_alert()
                .unwrap();
            Err(error)
        }
    }
}

fn exec_button(select: &str) {
    match exec_script(select) {
        Ok(_) => (),
        Err(error) => {
            MessageDialog::new()
            .set_type(MessageType::Error)
            .set_title("Erro")
            .set_text(&error.to_string())
            .show_alert()
            .unwrap();
        }
    }
}

fn search_button() -> Option<String> {
    let file_path = FileDialog::new().pick_file();

    if let Some(path) = file_path {
        let path_str = path.display().to_str();
        OPTIONS.lock().unwrap().push(path_str.clone());
        exec_button(&path_str);
        Some(path_str)
    } else {
        None
    }
}